arcaflow_plugin_fio/fio_schema.py

```diff
@@ -56,110 +62,443 @@ def is_sync(self) -> bool:
         return self.value in self._sync_io_engines
 
 
+duration_pattern_string = r"[1-9][0-9]*(?:d|h|m|s|ms|us)?"
+duration_pattern = re.compile(rf"^{duration_pattern_string}$")
+duration_range_pattern = re.compile(
+    rf"^{duration_pattern_string}(?:-{duration_pattern_string})?$"
+)
+
+size_pattern_string = (
+    r"0x[0-9a-fA-F]+|"
+    r"\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)?"
+)
+size_pattern = re.compile(
+    rf"^(?:{size_pattern_string})(?:,(?:{size_pattern_string})){{0,2}}$",
+    re.IGNORECASE,
+)
+
+size_pattern_with_percent_string = rf"{size_pattern_string}|[1-9][0-9]?%|100%"
```

Comment:
`size_pattern_with_percent_string` seems like an odd name when the first alternative doesn't have a "%"? Should this name be more general?

Reply:
I had the same stumble.  The answer is, "this is a size pattern, but one which allows percentages as well as absolute values"...so, my proposal to replace "with" with "or" didn't work, either.  🙂

Reply:
`size_or_percent_pattern`? Any alternative sounds bulky and awkward, unfortunately, but the "with percent" (when one alternative is not) seems a bit too misleading.

Reply:
If it settles your mind, the way to read `size_pattern_with_percent_string` is "a string containing the 'size pattern' with the option of a percent string added to it".  😏

Reply:
It doesn't exactly "settle my mind" -- but it's just an unclear name, and I guess not really worth any further effort.